pkg/app/pipedv1/plugin/kubernetes/deployment/server_test.go

```diff
@@ -168,3 +191,67 @@ func TestDeploymentService_executeK8sSyncStage(t *testing.T) {
 	assert.Equal(t, "0123456789", deployment.GetAnnotations()["pipecd.dev/commit-hash"])
 
 }
+
+func TestDeploymentService_executeK8sSyncStage_withInputNamespace(t *testing.T) {
+	ctx := context.Background()
+
+	// read the application config from the example file
+	cfg, err := os.ReadFile(filepath.Join(examplesDir(), "kubernetes", "simple", "app.pipecd.yaml"))
+	require.NoError(t, err)
+
+	// decode and override the autoCreateNamespace and namespace
+	spec, err := config.DecodeYAML[*kubeConfigPkg.KubernetesApplicationSpec](cfg)
+	require.NoError(t, err)
+	spec.Spec.Input.AutoCreateNamespace = true
+	spec.Spec.Input.Namespace = "test-namespace"
+	cfg, err = yaml.Marshal(spec)
+	require.NoError(t, err)
```

Comment:
override example config to ensure these fields are used when applying the resources